internal/gatewayapi/tls.go

```diff
@@ -96,3 +97,24 @@ func verifyHostname(cert *x509.Certificate, host *v1.Hostname) ([]string, error)
 
 	return nil, x509.HostnameError{Certificate: cert, Host: string(*host)}
 }
+
+func validateCertificate(data []byte) error {
```

Comment:
can we also insert this in https://github.com/envoyproxy/gateway/blob/c075d81d5a77fd72a7bda3b65c62c7bad49a7d1f/internal/gatewayapi/tls.go#L26, so we can validate server cert as well ?

Reply:
done